server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetAliasesAction.java

```diff
@@ -72,8 +72,15 @@ public String getName() {
         return "get_aliases_action";
     }
 
+    @Override
+    public boolean allowSystemIndexAccessByDefault() {
+        // If not given an index (only an alias), resolves `_all`, which causes deprecation warnings.
+        return true;
```

Comment:
Should resolution of _all that includes system indices trigger a deprecation warning? If we get all aliases and return system aliases such as `.security` in 7.x but stop doing so in 8.0, then that would be a breaking change right?

Reply:
> Should resolution of _all that includes system indices trigger a deprecation warning?

This does feel odd, but:
1) If it *doesn't* trigger a deprecation warning, then (for example) searches against `_all` that return documents in system indices will have an unannounced breaking change in 8.0, as you note in your second question.
2) As far as I can tell, in the presence of Security, it's impossible to tell in `IndexNameExpressionResolver` if the original query used a wildcard or not, as the original list of indices has by then been replaced. If I'm wrong and there's a good way to get that information, this might be a bit easier to fix.

> If we get all aliases and return system aliases such as .security in 7.x but stop doing so in 8.0, then that would be a breaking change right?

Sort of - if we go ahead with our plan of treating system indices created in 7.x with the previous 7.x behavior, then technically no, but if we upgrade all of the system indices shortly after creation, it'll feel like it to a user.

I'd definitely like to figure out something better to do here.

Reply:
> As far as I can tell, in the presence of Security, it's impossible to tell in IndexNameExpressionResolver if the original query used a wildcard or not, as the original list of indices has by then been replaced

That is 100% correct. I do wonder if we could have security place the original indices in the thread context and have the IndexNameExpressionResolver know that header/context key?

> if we go ahead with our plan of treating system indices created in 7.x with the previous 7.x behavior, then technically no, but if we upgrade all of the system indices shortly after ugprade, it'll feel like it to a user.

I think there may be confusion around treatment of 7.x system indices. The original plan involved every index that was dot-prefixed to be a system index except those created in 7.x or earlier. Once we made the decision to allow hidden indices to be dot-prefixed, that opened up the ability to no longer having to provide this layer of bwc for system indices since we could have user created indices, hidden indices, and system indices that all start with dots after an upgrade. Instead, we allow 7.x dot indices that are not system or hidden to behave normally. 

We're already marking specific indices as system indices in 7.x if a descriptor has been registered for that index to route the read operations to the correct thread pool.

I am wondering if the types removal behavior could fit here:

1. Implicit system index access without the parameter is deprecated in 7.x
2. Explicit system index access without the parameter is deprecated in 7.x
3. System index access with the parameter is not deprecated in 7.x
4. Implicit system index access without the parameter is a bad request in 8.0
5. Explicit system index access without the parameter is a bad request in 8.0
6. System index access with the parameter is deprecated in 8.0+
7. Parameter removed for 9.0 :)